README.md
NlsyLinks
================

NlsyLinks is a free downloadable R package to facilitate Behavior Genetic and Family Studies research using the NLSY samples (the National Longitudinal Survey of Youth).  These samples are based on a cross-generational longitudinal nationally representative sample of over 30,000 participants followed for up to 35 years.  There are almost 50,000 pairwise kinship links.

Get started at [here](http://www.bls.gov/nls/) the NLSY and [here](http://nlsy-links.github.io/NlsyLinks/) for the NlsyLinks package.
    

Installing
--------------------------

The *release* version of `NlsyLinks` can be installed from [CRAN](https://cran.r-project.org/package=NlsyLinks).

```r
install.packages("NlsyLinks")
```

The latest *development* version of `NlsyLinks` can be installed from [GitHub](https://github.com/nlsy-links/NlsyLinks/) after installing the `remotes` package.

```r
install.packages("remotes")
remotes::install_github(repo="nlsy-links/NlsyLinks")
```

Obtaining the Links
--------------------------

There are roughly three versions of the links:
1. The *release* versions are bundled in the `NlsyLinks` package and updated once or twice a year during development.
1. The *pre-release* versions are downloadable from our [team's website](http://nlsy-links.github.io/NlsyLinks/) and updated several times a month during development.  Formats include R, SAS, and plain-text CSV.
1. The *development* versions are downloadable from our [determination repo](https://github.com/nlsy-links/nlsy-links-determination-2017) updated several times a day during development.

## Locations for Help and Development

This repository contains the code used in the [NlsyLinks](https://cran.r-project.org/package=NlsyLinks) R package.  For additional information about the package and using it in NLSY research, please see http://nlsy-links.github.io/NlsyLinks/

[This GitHub repository](https://github.com/nlsy-links/NlsyLinksStaging) is our primary way of managing, tracking, and build-checking versions of the source code.  (We're no longer using our [R-Forge site](https://r-forge.r-project.org/projects/nlsylinks/).)  The stable releases (intended for most researchers) is available on the package's [CRAN site](https://cran.r-project.org/package=NlsyLinks).

We recently moved the forums to https://groups.google.com/forum/#!forum/nlsy-links.  Previous conversations are available at our old [R-Forge](https://r-forge.r-project.org/forum/?group_id=1330) and [Disqus](https://disqus.com/home/forum/nlsylinks/) sites.

Publication
--------------------------

Please see [research-publications](http://nlsy-links.github.io/NlsyLinks/research-publications.html) for a list of the 70+ publications arising from the kinship links.  An overview is available at:

Joseph Lee Rodgers, William H. Beasley, David E. Bard, Kelly M. Meredith, Michael D. Hunter, Amber B. Johnson, Maury Buster, Chengchang Li, Kim O. May, S. Mason Garrison, Warren B. Miller, Edwin van den Oord, and David C. Rowe (2016). *Behavior Genetics, 46*. [https://doi.org/10.1007/s10519-016-9785-3](https://doi.org/10.1007/s10519-016-9785-3).

>The National Longitudinal Survey of Youth datasets (NLSY79; NLSY-Children/Young Adults; NLSY97) have extensive family pedigree information contained within them. These data sources are based on probability sampling, a longitudinal design, and a cross-generational and within-family data structure, with hundreds of phenotypes relevant to behavior genetic (BG) researchers, as well as to other developmental and family researchers. These datasets provide a unique and powerful source of information for BG researchers. But much of the information required for biometrical modeling has been hidden, and has required substantial programming effort to uncover—until recently. Our research team has spent over 20 years developing kinship links to genetically inform biometrical modeling. In the most recent release of kinship links from two of the NLSY datasets, the direct kinship indicators included in the 2006 surveys allowed successful and unambiguous linking of over 94 % of the potential pairs. In this paper, we provide details for research teams interested in using the NLSY data portfolio to conduct BG (and other family-oriented) research.


Thanks to Funders
--------------------------

The current work on the NLSY Kinship links has been supported by NIH Grant R01-HD065865 ([Joe Rodgers](http://www.vanderbilt.edu/psychological_sciences/bio/joe-rodgers), PI).

Status
--------------------------

| [GitHub](https://github.com/nlsy-links/NlsyLinks) | [Travis-CI](https://travis-ci.org/nlsy-links/NlsyLinks/builds) | [AppVeyor](https://ci.appveyor.com/project/wibeasley/nlsylinks/history) |  [Coveralls](https://coveralls.io/r/nlsy-links/NlsyLinks) |
| :----- | :---------------------------: | :------------: | :-------: |
| [Master](https://github.com/nlsy-links/NlsyLinks/tree/master) |  [![Travis-CI Build Status](https://travis-ci.org/nlsy-links/NlsyLinks.png?branch=master)](https://travis-ci.org/nlsy-links/NlsyLinks) | [![Build status](https://ci.appveyor.com/api/projects/status/fo1oeqn9734dhhmu/branch/master?svg=true)](https://ci.appveyor.com/project/wibeasley/nlsylinks/branch/master) | [![Coverage Status](https://coveralls.io/repos/nlsy-links/NlsyLinks/badge.svg?branch=master)](https://coveralls.io/r/nlsy-links/NlsyLinks?branch=master) |
| [Dev](https://github.com/nlsy-links/NlsyLinks/tree/dev) | [![Travis-CI Build Status](https://travis-ci.org/nlsy-links/NlsyLinks.png?branch=dev)](https://travis-ci.org/nlsy-links/NlsyLinks) | [![Build status](https://ci.appveyor.com/api/projects/status/fo1oeqn9734dhhmu/branch/dev?svg=true)](https://ci.appveyor.com/project/wibeasley/nlsylinks/branch/dev) | [![Coverage Status](https://coveralls.io/repos/nlsy-links/NlsyLinks/badge.svg?branch=dev)](https://coveralls.io/r/nlsy-links/NlsyLinks?branch=dev) |
| | *Ubuntu LTS* | *Windows Server* |  *Travis Code Coverage* |


| [CRAN](https://cran.r-project.org/) | [Version](https://cran.r-project.org/package=NlsyLinks) | [Rate](http://cranlogs.r-pkg.org/) | [Zenodo](https://zenodo.org/search?ln=en&p=nlsylinks) |
|  :---- | :----: | :----: | :----: |
| [Latest](https://cran.r-project.org/package=NlsyLinks) | [![CRAN_Status_Badge](http://www.r-pkg.org/badges/version/NlsyLinks)](https://cran.r-project.org/package=NlsyLinks) | ![CRANPace](http://cranlogs.r-pkg.org/badges/NlsyLinks) | [![DOI](https://zenodo.org/badge/doi/10.5281/zenodo.49941.svg)](http://dx.doi.org/10.5281/zenodo.49941) |
|   | *Latest CRAN version* | *CRAN Downloads* | *Independently-hosted Archive* |


NEWS


Version 2.0.8 (2018-?-?)
------------------------------------------------------------------------------
Updates:
* [pkgdown](https://hadley.github.io/pkgdown/) documentation
* Rename dataset `SurveyDate` to `Survey79`.


Version 2.0.5 (2016-03-17)
------------------------------------------------------------------------------
Updates:
* Small updates to meet new testthat version.

Version 2.0.1 (2015-08-16)
------------------------------------------------------------------------------
Updates:
* SurveyDate dataset contains survey details for each subject, for each year.
* Small updates to meet newer CRAN guidelines.

Version 1.4005 (2015-01-15)
------------------------------------------------------------------------------
Updates:
* Small updates to vignettes and documentation.
* Small updates to meet newer CRAN guidelines about title & description format.
* Checked with R Tools 3.2.

Version 1.300 (2013-11-21)
------------------------------------------------------------------------------
New Features:
* A more complete version of the Gen1 links are included.  95% of all Gen1 Housemates have been linked.

Updates:
* `R` now has missing values for Gen1 pairs if the 1979 roster (eg, R00001.51) doesn't indicate they're the same generation.  If this value is desirable for your research, their R values are still available in `RFull`.
* The naming scheme for the `Links79Pair` and `Links79PairExpanded` datasets are more consistent across variables.  For variables that are measured separately for both subjects (eg, Gender), the subjects' variable name will have an `_S1` or `_S2` appended to it.  For instance, the variables `LastSurvey_S1` and `LastSurvey_S2` correspond to the last surveys completed by the pair's first and second subject, respectively.
* Similarly, the functions `CreatePairLinksDoubleEntered` and `CreatePairLinksSingleEntered` now by default append `_S1` and `_S2`, instead of `_1` and `_2`.  However this can be modified using the `subject1Qualifier` and `subject2Qualifier` parameters.
* Added function `ReadCsvNlsy79Gen1()`
* Updated reference manual, ACE vignette, and FAQ.

Version 1.200 (2012-12-18)
------------------------------------------------------------------------------
Updates:
* Renamed `R` to `RFull`.  `RFull` contains all the _R_ values we have.  `R` excludes _R_ values in Gen1Housemates if they're 0, .375, or .75.  The associated relationships (ie, Gen2Cousins and AuntNieces) become null for _R_, but not for _RFull_.

Version 1.017 (2012-11-29)
------------------------------------------------------------------------------
Updates:
* Removed code that assigned all unassigned siblings to .375.  This doesn't make sense for Gen1, and was mostly an internal experiment (and wasn't released on CRAN).

Version 1.015 (2012-11-10)
------------------------------------------------------------------------------
Updates:
* Added outcome variables for Gen1.  Renamed some outcome variables for Gen2.

Version 1.014 (2012-11-09)
------------------------------------------------------------------------------
Updates:
* Based on geocode variables, five Gen1 ambiguous twin was demoted to R=0 (two were R=.5; three were R=NULL).

Version 1.013 (2012-11-08)
------------------------------------------------------------------------------
Updates:
* Rebuilt For R2.15.2 with with RTools216.exe.

Version 1.012 (2012-10-11)
------------------------------------------------------------------------------
Updates:
* Small documentation updates

Version 1.011 (2012-10-11)
------------------------------------------------------------------------------
Updates:
* Gen1 Links are more thorough (but still in a beta state)
* Gen1 and Gen2 information uses the NLSY 2010 survey wave (released this Spring 2012)

Version 1.010 (2012-10-11)
------------------------------------------------------------------------------
* Refreshing after development interrupt to make sure everything builds and tests correctly

Version 1.007 (2012-06-08)
------------------------------------------------------------------------------
Updates:
* Added material to appendices of the NlsyAce vignette.
* Removed 'stringr' package from Depends list.  It's no longer used creating lavaan syntax.

Version 1.004 (2012-06-07)
------------------------------------------------------------------------------
New Features:
* Added ParentChild links.
* Added Gen1 links (using only Explicit items).  There will be much more work to these relationship in the next few months.
* Added Gen2Cousins & AuntNieces.  These will change as the Gen1 links are filled out with implicit information.

Version 1.003 (2012-05-21)
------------------------------------------------------------------------------
Updates:
* Additional Documentation.

Version 1.001 (2012-05-04)
------------------------------------------------------------------------------
Updates:
* Started a FAQ (as a vignette).

Version 1.0 (2012-04-27)
------------------------------------------------------------------------------
Updates:
* Additional documentation and unit tests.
* Changed R-Forge development status from "Beta" to "Production/Stable"

Version 0.29 (2012-04-22)
------------------------------------------------------------------------------
Updates:
* Modified some function signatures to be make package more consistent.
* Additional lavaan example in ACE vignette.

Version 0.26 (2012-04-20)
------------------------------------------------------------------------------
New Features:
* The new function 'AceLavaanGroup' uses the 'lavaan' package to estimate a multigroup SEM ACE model.

Version 0.24 (2012-04-16)
------------------------------------------------------------------------------
New Features:
* The S4 class 'AceEstimate' is returned by the 'Ace' function.
* Methods & functions associated with the AceEstimate class.
* SubjectDetails79 dataset for Gen2.
* ReadCsvNlsy79 function.
* Validation functions: (a) OutComeDataset
* Column Utilities: (a) VerifyColumnExists, (c) RenameNlsyColumn


R/nlsy-links.R
#' @title Utilities and kinship information for Behavior Genetics and Developmental research using the NLSY.
#'
#' @description Utilities and kinship information for Behavior Genetics and Developmental research using the NLSY.
#' Researchers and grad students interested using the [NLSY](http://www.bls.gov/nls/) for Behavior Genetics and family research, please start with our 2016 article,
#' '[*The NLSY Kinship Links: Using the NLSY79 and NLSY-Children Data to Conduct Genetically-Informed and Family-Oriented Research*](https://link.springer.com/article/10.1007/s10519-016-9785-3).
#'
#' @docType package
#'
#' @name NlsyLinks-package
#'
#' @aliases NlsyLinks
#'
#' @note This package considers both Gen1 and Gen2 subjects.  "Gen1" refers to
#' subjects in the original NLSY79 sample
#' (http://www.bls.gov/nls/nlsy79.htm).  "Gen2" subjects are the
#' biological children of the Gen1 females -ie, those in the NLSY79 Children
#' and Young Adults sample (http://www.bls.gov/nls/nlsy79ch.htm).
#'
#' The release version is available through [CRAN](https://cran.r-project.org/package=NlsyLinks) by
#' running `install.packages('NlsyLinks')`.
#' The most recent development version is available through [GitHub](https://github.com/nlsy-links/NlsyLinks) by
#' running
#' `devtools::install_github` `(repo = 'nlsy-links/NlsyLinks')`
#' (make sure [devtools](https://cran.r-project.org/package=devtools) is already installed).
#' If you're having trouble with the package, please install the development version.  If this doesn't solve
#' your problem, please create a [new issue](https://github.com/nlsy-links/NlsyLinks/issues), or email Will.
#'
#' @author
#' [William Howard Beasley](http://scholar.google.com/citations?user=ffsJTC0AAAAJ) ([Howard Live Oak LLC](http://howardliveoak.com/), Norman)
#'
#' [Joseph Lee Rodgers](http://www.vanderbilt.edu/psychological_sciences/bio/joe-rodgers) (Vanderbilt University, Nashville)
#'
#' [David Bard](http://find.ouhsc.edu/Faculty.aspx?FacultyID=1041) (University of Oklahoma Health Sciences Center, OKC)
#'
#' Kelly Meredith (Oklahoma City University, OKC)
#'
#' [Michael D. Hunter](http://mhuntero.com/) (University of Oklahoma, Norman)
#'
#' Maintainer: Will Beasley <wibeasley@@hotmail.com>
#'
#' @references This package's development was largely supported by the NIH
#' Grant 1R01HD65865, ["NLSY Kinship Links: Reliable and Valid Sibling
#' Identification"](http://taggs.hhs.gov/AwardDetail.cfm?s_Award_Num=R01HD065865&n_Prog_Office_Code=50)
#' (PI: Joe Rodgers).  A more complete list of research articles
#' using NLSY Kinship Links is maintained on our [package's
#' website](http://nlsy-links.github.io/NlsyLinks/research-publications.html).
#'
#' Rodgers, Joseph Lee, & Kohler, Hans-Peter (2005).
#' [Reformulating and simplifying the DF analysis model](http://www.springerlink.com/content/n3x1v1q282583366/).
#' *Behavior Genetics, 35* (2), 211-217.
#'
#' Rodgers, J.L., Bard, D., Johnson, A., D'Onofrio, B., & Miller, W.B. (2008).
#' [The Cross-Generational Mother-Daughter-Aunt-Niece Design: Establishing
#' Validity of the MDAN Design with NLSY Fertility Variables.](http://www.ncbi.nlm.nih.gov/pubmed/18825497).
#' *Behavior Genetics, 38*, 567-578.
#'
#' D'Onofrio, B.M., Van Hulle, C.A., Waldman, I.D., Rodgers, J.L., Rathouz,
#' P.J., & Lahey, B.B. (2007). [Causal inferences regarding prenatal alcohol
#' exposure and childhood externalizing problems.](http://www.ncbi.nlm.nih.gov/pubmed/17984398).
#' *Archives of General Psychiatry, 64*, 1296-1304.
#'
#' Rodgers, J.L. & Doughty, D. (2000).  [Genetic and environmental influences
#' on fertility expectations and outcomes using NLSY kinship data.](http://link.springer.com/chapter/10.1007\%2F978-1-4615-4467-8_6)
#' In J.L. Rodgers, D. Rowe, & W.B. Miller (Eds.)
#' *Genetic influences on fertility and sexuality.* Boston: Kluwer Academic Press.
#'
#' Cleveland, H.H., Wiebe, R.P., van den Oord, E.J.C.G., & Rowe, D.C. (2000).
#' [Behavior problems among children from different family structures: The
#' influence of genetic self-selection.](http://www.ncbi.nlm.nih.gov/pubmed/10953940)
#' *Child Development, 71*, 733-751.
#'
#' Rodgers, J.L., Rowe, D.C., & Buster, M. (1999). [Nature, nurture, and first
#' sexual intercourse in the USA: Fitting behavioural genetic models to NLSY
#' kinship data](http://www.ncbi.nlm.nih.gov/pubmed/10081235).
#' *Journal of Biosocial Sciences, 31*.
#'
#' Rodgers, J.L., Rowe, D.C., & Li, C. (1994). [Beyond nature versus nurture:
#' DF analysis of nonshared influences on problem behaviors.](http://psycnet.apa.org/journals/dev/30/3/374/)
#' *Developmental Psychology, 30*, 374-384.
#'
#' @keywords package
#'
#' @examples
#' library(NlsyLinks)    # Load the package into the current R session.
#' summary(Links79Pair)  # Summarize the five variables.
#' hist(Links79Pair$R)   # Display a histogram of the Relatedness values.
#' table(Links79Pair$R)  # Create a table of the Relatedness values for the whole sample.
#'
#' \dontrun{
#' # Install/update NlsyLinks with the release version from CRAN.
#' install.packages('NlsyLinks')
#'
#' # Install/update NlsyLinks with the development version from GitHub
#' #install.packages('devtools') #Uncomment if 'devtools' isn't installed already.
#' devtools::install_github('nlsy-links/NlsyLinks')
#' }
NULL
